src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py

```diff
@@ -918,7 +917,16 @@ def __call__(
         if isinstance(self.controlnet, MultiControlNetModel) and isinstance(controlnet_conditioning_scale, float):
             controlnet_conditioning_scale = [controlnet_conditioning_scale] * len(self.controlnet.nets)
 
-        # 3. Encode input prompt
+        # 3. Determination of whether to infer ControlNet using only for the conditional batch.
```

Comment:
I'd say we actually just always set `guess_mode` to True whenever global_pool_conditions is True.

E.g.:

```py
global_pool_conditions = self.controlnet.config.global_pool_conditions if isinstance(self.controlnet, ControlNetModel) else self.controlnet.nets[0].config.global_pool_conditions
guess_mode = guess_mode or global_pool_conditions